### PR TITLE
[dv/otp_ctrl] Update testplan for V2 review

### PR DESCRIPTION
**Pull request by cindychip**
This PR updates the otp_ctrl's testplan for V2 review.

Signed-off-by: Cindy Chen <chencindy@opentitan.org>

### CODE REVIEW

**Review by udinator — APPROVED** [review]

File: hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson [hunk 1]
```diff
@@ -96,16 +83,35 @@
...
-            which isn't tested in common CSR tests. HW will set this register to 0 when OTP_CTRL's
-            DAI interface is busy.
+            The `direct_access_regwen` is RO register and it gates bunch of write access of other
+            registers, which isn't tested in common CSR tests. HW will set this register to 0 when
+            OTP_CTRL's DAI interface is busy.
 
             Stimulus and checks:
-            - Random read DIRECT_ACCESS_REGWEN and ensure the value is set to 0 during OTP init.
-            - Test accessing registers gated by DIRECT_ACCESS_REGWEN is ignored during OTP init.
+            - Random read `direct_access_regwen` and ensure the value is set to 0 during OTP init.
```
Comment (sriyerg) [inline hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson:113]:
```suggestion
            - Randomly read `direct_access_regwen` and verify that it returns 0 during OTP init.
```

File: hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson [hunk 2]
```diff
@@ -96,16 +83,35 @@
...
+            - Test accessing registers gated by `direct_access_regwen` is ignored during OTP init.
```
Comment (sriyerg) [inline hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson:114]:
```suggestion
            - Verify that the writes to the registers controlled by it do not go through during OTP init. 
```

File: hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson [hunk 3]
```diff
@@ -133,22 +139,27 @@
     {
       name: lc_interactions
       desc: '''
-            This test check otp and life_cycle interactions. Based on the DAI access sequence,
+            This test check OTP_CTRL and LC_CTRL interactions. Based on the DAI access sequence,
```
Comment (sriyerg) [inline hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson:142]: All of these are tests, so we can jump straight to what its goal is:
```suggestion
            Verify the OTP_CTRL and LC_CTRL interactions. Based on the DAI access sequence,
```

File: hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson [hunk 4]
```diff
@@ -133,22 +139,27 @@
...
             this test will run the following sequences in parallel:
```
Comment (sriyerg) [inline hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson:143]:
```suggestion
            run the following sequences in parallel:
```

**Comment by cindychip** [comment]
Thanks Sri for all the suggestions! I updated this PR accordingly and fixes the inconsistency in the previous version.

File: hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson [hunk 5]
```diff
@@ -60,52 +61,63 @@
...
+            - write to a random OTP partition after digest is issued but before reset is asserted
 
-            If OTP init finished without fatal alert, this test will check:
-            - OTP init finishes with power init output goes to 1
-            - Status and interrupt registers reflect the correct ecc correctable error
+            **Checks**:
+            - check if the corresponding alerts are triggered
+            - check if the error_code register is set correctly
+            Note that due to limited simulation time, for background checks, this test only write
+            random value that is less than 'd20 to the check period.
```
Comment (matutem) [inline hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson:104]: 'd20 -> 20

File: hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson [hunk 6]
```diff
@@ -60,52 +61,63 @@
...
+            Create a failure scenario by randomly picking one of these three methods:
+            - inject ECC errors into the OTP macro via backdoor
+            - set the `check_timeout` CSR with a very small value
+            - write to a random OTP partition after digest is issued but before reset is asserted
 
-            If OTP init finished without fatal alert, this test will check:
-            - OTP init finishes with power init output goes to 1
-            - Status and interrupt registers reflect the correct ecc correctable error
+            **Checks**:
+            - check if the corresponding alerts are triggered
```
Comment (matutem) [inline hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson:101]: Remove "check if", since this is a list of checks.

**Review by cindychip — COMMENTED** [review]
Thanks Sri for the detailed comments. I believe I updated them all according to your suggestions!

File: hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson (hunk 6 above)
Comment (cindychip) [reply hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson]: Thanks Matute. Yup, I removed the check if.

File: hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson (hunk 5 above)
Comment (cindychip) [reply hw/ip/otp_ctrl/data/otp_ctrl_testplan.hjson]: Thanks, updated

**Review by udinator — APPROVED** [review]
lgtm in general

**Review by matutem — APPROVED** [review]
LGTM

**Pull request merged**